src/benchmark/file_io_write_micro_benchmark.cpp

```diff
@@ -150,6 +184,92 @@ void FileIOWriteMicroBenchmarkFixture::pwrite_atomic_multi_threaded(benchmark::S
   }
 }
 
+void FileIOWriteMicroBenchmarkFixture::aio_single_threaded(benchmark::State& state) {
+  auto fd = int32_t{};
+  if ((fd = open(filename, O_WRONLY)) < 0) {
+    close(fd);
+    Fail("Open error:" + std::strerror(errno));
+  }
+
+  for (auto _ : state) {
+    state.PauseTiming();
+    micro_benchmark_clear_disk_cache();
+    state.ResumeTiming();
+
+    auto* data_to_write_start = std::data(data_to_write);
+
+    // The standard aio control block. A structure, that holds the information about the asnyc IO op.
+    struct aiocb aiocb;
+    memset(&aiocb, 0, sizeof(struct aiocb));
+    aiocb.aio_fildes = fd;
+    aiocb.aio_offset = 0;
+    aiocb.aio_buf = data_to_write_start;
+    aiocb.aio_nbytes = NUMBER_OF_BYTES;
+    aiocb.aio_lio_opcode = LIO_WRITE;
+
+    Assert(aio_write(&aiocb) != -1, "Read error: " + std::strerror(errno));
+
+    auto err = aio_error(&aiocb);
+    /* Wait until end of transaction */
+    while (err == EINPROGRESS) {
+      err = aio_error(&aiocb);
+    }
+
+    aio_error_handling(&aiocb, NUMBER_OF_BYTES);
+
+    state.PauseTiming();
+    sanity_check();
+    state.ResumeTiming();
+  }
+
+  close(fd);
+}
+
+void FileIOWriteMicroBenchmarkFixture::aio_multi_threaded(benchmark::State& state, uint16_t thread_count) {
+  auto filedescriptors = std::vector<int32_t>(thread_count);
+  for (auto i = size_t{0}; i < thread_count; i++) {
```

Comment:
```suggestion
  for (auto index = size_t{0}; index < thread_count; ++index) {
```

please change also in line 236

Reply:
The whole file is with 'i' instead of index, should we change each loop?